src/pages/institution/institution-solution/institution-solution.ts

```diff
@@ -0,0 +1,266 @@
+import { Component } from "@angular/core";
+import { NavController, NavParams } from "ionic-angular";
+import { LocalStorageProvider } from "../../../providers/local-storage/local-storage";
+import { UtilsProvider } from "../../../providers/utils/utils";
+import { EvidenceProvider } from "../../../providers/evidence/evidence";
+import { InstitutionServiceProvider } from "../institution-service";
+import { storageKeys } from "../../../providers/storageKeys";
+import { ProgramObservationSubmissionPage } from "../../programs/program-observation-submission/program-observation-submission";
+import { ProgramServiceProvider } from "../../programs/program-service";
+
+/**
+ * Generated class for the InstitutionSolutionPage page.
+ *
+ * See https://ionicframework.com/docs/components/#navigation for more info on
+ * Ionic pages and navigation.
+ */
+
+@Component({
+  selector: "page-institution-solution",
+  templateUrl: "institution-solution.html",
+})
+export class InstitutionSolutionPage {
+  entity: any;
+  entityType: any;
+  entityIndex: any;
+  institutionList: any;
+  submissionArr: any;
+
+  constructor(
+    public navCtrl: NavController,
+    public navParams: NavParams,
+    public localStorage: LocalStorageProvider,
+    public utils: UtilsProvider,
+    public evdnsServ: EvidenceProvider,
+    public institutionService: InstitutionServiceProvider,
+    public programService: ProgramServiceProvider
+  ) {}
+
+  ionViewDidLoad() {
+    console.log("ionViewDidLoad InstitutionSolutionPage");
+    let navData = this.navParams.get("navData");
+    this.entityType = navData.entityType;
+    this.entityIndex = navData.entityIndex;
+    this.getInstituionFromStorage();
+  }
+
+  getInstituionFromStorage(stopLoader?) {
+    stopLoader ? null : this.utils.startLoader();
+    this.institutionService
+      .getInstituionFromStorage()
+      .then((institutions) => {
+        this.institutionList = institutions;
+        this.entity = this.institutionList.entities[this.entityType][
+          this.entityIndex
+        ];
+        this.getSubmissionArr();
+        this.utils.stopLoader();
+      })
+      .catch((error) => {
+        this.entity = null;
+        this.utils.stopLoader();
+      });
+  }
+
+  getSubmissionArr() {
+    this.localStorage
+      .getLocalStorage(storageKeys.submissionIdArray)
+      .then((allId) => {
+        this.submissionArr = allId;
+        this.applySubmission();
+      })
+      .catch((err) => {});
+  }
+
+  applySubmission() {
+    let entityId = this.institutionList.entities[this.entityType][
+      this.entityIndex
+    ]._id;
+
+    this.institutionList.entities[this.entityType][
+      this.entityIndex
+    ].solutions.map((sol, solutionIndex) => {
+      this.submissionArr.includes(sol.submissionId)
+        ? (sol.downloaded = true)
+        : null;
+    });
+  }
+
+  goToEcm(solutionIndex, submissionId) {
```

Comment:
Can we reuse this in programs and the institution's flow?

Reply:
same reason as the structure is different i am using two different functions to manipulate the data in each flows